Lib/test/test_urlparse.py

```diff
@@ -1402,16 +1402,34 @@ def test_issue14072(self):
         self.assertEqual(p2.path, '+31641044153')
 
     def test_invalid_bracketed_hosts(self):
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[192.0.2.146]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[important.com:8000]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[v123r.IP]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[v12ae]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[v.IP]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[v123.]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[v]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[0439:23af::2309::fae7:1234]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@[0439:23af:2309::fae7:1234:2342:438e:192.0.2.146]/Path?Query')
-        self.assertRaises(ValueError, urllib.parse.urlsplit, 'Scheme://user@]v6a.ip[/Path')
+        cases = [
+            'Scheme://user@[192.0.2.146]/Path?Query',
+            'Scheme://user@[important.com:8000]/Path?Query',
+            'Scheme://user@[v123r.IP]/Path?Query',
+            'Scheme://user@[v12ae]/Path?Query',
+            'Scheme://user@[v.IP]/Path?Query',
+            'Scheme://user@[v123.]/Path?Query',
+            'Scheme://user@[v]/Path?Query',
+            'Scheme://user@[0439:23af::2309::fae7:1234]/Path?Query',
+            'Scheme://user@[0439:23af:2309::fae7:1234:2342:438e:192.0.2.146]/Path?Query',
+            'Scheme://user@]v6a.ip[/Path',
+            'Scheme://user@[v6a.ip/path?query',
+            'Scheme://user@v6a.ip]/path?query',
+            'Scheme://user@prefix.[v6a.ip]/path?query',
+            'Scheme://user@[v6a.ip].suffix/path?query',
+        ]
+
+        for case in cases:
+            with self.subTest(case=case):
+                with self.assertRaises(ValueError):
+                    urllib.parse.urlsplit(case).hostname
```

Comment:
Previously urlsplit raised the ValueError. This PR as is changes that and only raises it upon hostname or port attribute access.

We need to continue raising the ValueError at urlsplit time. Existing code must be assumed to expect that split time validation behavior rather than at attribute access time.

Reply:
Thanks, @gpshead. I updated the code to do the check at urlsplit time. (I also pushed another tiny commit, but GitHub seems to be stuck processing that update, and it's not showing up.)